lib/internal/streams/readable.js

```diff
@@ -290,14 +289,16 @@ function Readable(options) {
     }
   });
 
+  const isObjectMode = this._readableState.objectMode;
+
   // Manually shove something into the read() buffer.
   // This returns true if the highWaterMark has not been hit yet,
   // similar to how Writable.write() returns true if you should
   // write() some more.
-  this.push = this._readableState.objectMode ? readableAddChunkPushObjectMode : readableAddChunkPushByteMode;
+  this.push = isObjectMode ? readableAddChunkPushObjectMode : readableAddChunkPushByteMode;
 
   // Unshift should *always* be something directly out of read().
-  this.unshift = this._readableState.objectMode ? readableAddChunkUnshiftObjectMode : readableAddChunkUnshiftByteMode;
+  this.unshift = isObjectMode ? readableAddChunkUnshiftObjectMode : readableAddChunkUnshiftByteMode;
```

Comment:
`construct`  might call push and unshift.

Reply:
reverted to have prototype function as before